capella_model_explorer/app.py

```diff
@@ -176,13 +176,31 @@ def rendered_report(template_id: str, model_element_uuid: str = "") -> t.Any:
             ),
             cls="dark:text-neutral-100 grow content-center",
         )
+
+    processed_html, toc_items = reports.process_html_with_toc(
+        rendered_template
+    )
+
+    toc_button_script = ""
+    if toc_items:
+        toc_button_script = (
+            "const tocButton = document.getElementById('toc-toggle-button');"
+            "if (tocButton) { tocButton.classList.remove('hidden'); "
+            "tocButton.classList.add('xl:hidden'); }"
+        )
```

Comment:
If we:

- always generate the `#toc-toggle-button`
- always set `xl:hidden` on it (in addition to the initial "hidden")

Then we can reduce this to a simple `tocButton.classList.remove('hidden');` without any other conditions/checks.